services/src/main/java/org/keycloak/organization/admin/resource/OrganizationResource.java

```diff
@@ -152,6 +154,14 @@ private OrganizationModel toModel(OrganizationRepresentation rep, OrganizationMo
 
         model.setName(rep.getName());
 
+        if (rep.getAttributes() != null) {
+            Set<String> attrsToRemove = model.getAttributes().keySet();
```

Comment:
One thing that we have in other places that I think we can start improve here is to encapsulate how attributes are set to a model.

This logic can be easily encapsulated in the domain model so that we don't need to repeat ourselves every time we need to update/set attributes to an organization.

wdyt?

Reply:
I'll work on it as a follow-up.